syft/frameworks/torch/tensors/interpreters/replicated_shared.py

```diff
@@ -270,6 +273,98 @@ def shape(self):
     def players(self):
         return self.__get_players()
 
+    def set_garbage_collect_data(self, value):
+        shares = self.child
+        for _, share in shares.items():
+            share[0].garbage_collect_data = value
+            share[1].garbage_collect_data = value
+
+    def get_garbage_collect_data(self):
+        shares = self.child
+        res = {}
+
+        for worker, shares in shares.items():
+            assert shares[0].garbage_collect_data == shares[1].garbage_collect_data
+            res[worker.id] = shares[0].garbage_collect_data
+
+        return res
```

Comment:
I wonder if all shares should not always have the same `garbage_collect_data` ?
If so, you can put the assert all equal test on all the shares attributes

Reply:
I can put for all the shares to have the same garbage collect data :D